src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemProcessorStateWriter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using ProductConstructionService.Common;
+
+namespace ProductConstructionService.WorkItems;
+public class WorkItemProcessorStateWriter
```

Comment:
StateManager?

Reply:
But it doesn't really manage it, it just changes and fetches it

Reply:
But I guess Writer isn't very telling of it's ability to fetch it too, it might be better, the StateManager

Reply:
Or maybe StateCache?

Reply:
yes! I think that's the best one